libs/MeshKernel/src/Polygon.cpp

```diff
@@ -349,21 +349,78 @@ std::tuple<double, meshkernel::Point, meshkernel::TraversalDirection> meshkernel
     const Point reference = ReferencePoint(polygon, projection);
     const auto numberOfPointsOpenedPolygon = static_cast<UInt>(polygon.size()) - 1;
 
-    for (UInt n = 0; n < numberOfPointsOpenedPolygon; ++n)
+    if (numberOfPointsOpenedPolygon == 3)
```

Comment:
i think we have defined constants in mesh2d

        static constexpr UInt m_numNodesQuads = 4;                                                 ///< Number of nodes in a quadrilateral
        static constexpr UInt m_numNodesInTriangle = 3;                                            ///< Number of nodes in a triangle

maybe we should move these constants to meshkernel::constants namespace

Reply:
I made this change, but in another item